From 975b53677da0f0da29c1f55a0479d7982de8a8c7 Mon Sep 17 00:00:00 2001
From: "Dr. Greg" <greg@enjellic.com>
Date: Sat, 20 Feb 2021 09:38:27 -0600
Subject: [PATCH 31/50] Modify diagnostic information for inode security drop.

Previously the deletion of an inode caused two diagnostic
messages to be output with pr_info status.  This was extraneous
information, particularly during system initialization.  The
changes to the kernel to shroud pointer addresses also made one
of the message not useful.

This update drops the display of the address of the inode and
moves to a single diagnostic message indicating that a security
inode had been dropped.
---
 security/integrity/iint.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/security/integrity/iint.c b/security/integrity/iint.c
index 6ff1b9950927..8668c534ddf7 100644
--- a/security/integrity/iint.c
+++ b/security/integrity/iint.c
@@ -141,17 +141,16 @@ void integrity_inode_free(struct inode *inode)
 
 	if (!IS_IMA(inode))
 		return;
-	pr_info("[%s]: Processing eviction for inode=%ld, flags=%u\n",
-		__func__, inode->i_ino, inode->i_flags);
 
 	write_lock(&integrity_iint_lock);
 	iint = __integrity_iint_find(inode);
 	rb_erase(&iint->rb_node, &integrity_iint_tree);
 	write_unlock(&integrity_iint_lock);
 
-	pr_info("[%s]: Evicting security inode %p for %ld.\n", __func__,
-		iint, iint->inode->i_ino);
 	iint_free(iint);
+
+	pr_debug("[%s]: Processed IMA eviction for inode=%ld, flags=%u\n",
+		 __func__, inode->i_ino, inode->i_flags);
 	inode->i_flags &= ~S_IMA;
 }
 
-- 
2.31.1

